Detour/Include/DetourModernCpp.h

```diff
@@ -0,0 +1,24 @@
+#ifndef DETOURMODERNCPP_H
+#define DETOURMODERNCPP_H
+
+#include <limits>
+
+#include "DetourAssert.h"
+
+#if defined(__cplusplus) && __cplusplus >= 201103L
+#define DT_NULL nullptr
+#define DT_OVERRIDE override
+#else
+#include <stddef.h>
+#define DT_NULL NULL
+#define DT_OVERRIDE
+#endif
+
+template <typename TO, typename FROM>
+inline TO dtCast( FROM val )
+{
+	dtAssert( std::numeric_limits<TO>::min() <= val && val <= std::numeric_limits<TO>::max() );
+	return static_cast<TO>( val );
+}
```

Comment:
This doesn't seem only applicable to "modern C++", so I'd be cool moving this to a `DetourCommon.h`

Also, I think a name like `dtSafeNumericCast` or `dtCheckedNumericCast` would be more appropriate here since this only works for numeric types, and it's checking that the value can be casted to the new type without losing information.